sysdig/data_source_sysdig_fargate_workload_agent.go

```diff
@@ -36,6 +36,13 @@ const agentinoKiltDefinition = `build {
     ]
 }`
 
+var (
+	kiltTagKeys = []string{
```

Comment:
Open question - should we allow any arbitrary tags or is it ok to be explicit here?

Reply:
TBH I'd add a separate `ignore_containers` setting that took a list of container names, instead of tightly following the CloudFormation interface. Also, I'm not exactly sure what the use case for ignoring all containers is: just don't pass the task through our instrumentation instead? But if we do want to preserve it, I'd support it as a boolean flag (ignore_all = true/false).

We could then build the tags map ourselves and would have a place to actually add useful documentation (I'm not sure "kilt recipe tags" will be informative to anybody, us after a few months included). It would also sidestep the question you asked here entirely ;)

Reply:
My main aim was to exactly duplicate the CF interface for this since the documentation is already there an the customer in question tried it. I will ask PM to see what they would prefer, code-wise it's a small delta to change the tags. I think this answers your very valid question about the "ignore all" option, as it is a slightly round-about way to disable instrumentation of the entire task.

Reply:
@gnosek I spoke to PM - they are happy to copy CF functionality to unblock the customer. I will raise an improvement ticket to review the TF API.

Reply:
I'm not sure it's a good idea to introduce a new feature just to change it again soon. IMHO let's either do it the TF way from the beginning or just leave the internals exposed. Otherwise we'll have to maintain two sets of docs and won't have a simple answer for "so how do I ignore a container in tf?" "well, it depends on your tf provider version"

Reply:
I'd avoid defining a data struct at the Terraform-level, and rather keep the info related to the container to be ignored at the TaskDefinition-level, as we already do in CFN.
That's for two main reasons:
 - That info relates to the design of the TaskDefinition more than to its deployment.
 - Different containers on different TaskDefinitions could have the same name, so the parameter at the Terraform-level should be a map to identify uniquely the containers to be skipped. In case of a massive deployment that wouldn't be practical.

Reply:
I'm not advocating any changes to how the process works, just an interface change from:
```
kilt_tags = {
  kilt_ignore_containers = "foo:bar"
}
```

to
```
ignore_containers = ["foo", "bar"]
```

(modulo exact hcl syntax).

This shouls all live in the kilt data source. If you need different ignored lists for different workloads, you need specific data sources anyway (there's nowhere to put the tags in the container definition json, since it lives at a higher level).

Note how https://registry.terraform.io/modules/cn-terraform/ecs-fargate-task-definition/aws/latest splits out all the top-level items of the task definition into separate terraform-level variables and we only get the container definition to work with

Reply:
I can rename my "API" to simplify it, like this simplified example:
```
data "sysdig_fargate_workload_agent" "containers_instrumented" {
  container_definitions = jsonencode([
    {
      "name" : "nginx",
      "image" : "nginx:latest",
...
    ignore_containers = ["nginx"]
  }
}

resource "aws_ecs_task_definition" "task_definition" {
  family             = "${var.prefix}-instrumented-task-definition"
  requires_compatibilities = ["FARGATE"]

  container_definitions = data.sysdig_fargate_workload_agent.containers_instrumented.output_container_definitions
}
```

As @gnosek points out, our data provider does not have the scope to define the "tags" within the task definition itself. I am very happy to drop "support" for the `kilt-ignore` tag as it really does not make much sense here.